File: DataSets/DataSet-Old/LeetCodeTasksDebug/Fixed/92_reverse-linked-list-ii/solutions/java/solution.java
class Solution {

    public ListNode reverseBetween(ListNode head, int left, int right) {

        if (head == null || left == right) return head;

        

        ListNode dummy = new ListNode(0);  // Dummy node to handle edge cases

        dummy.next = head;

        ListNode prev = dummy;

        

        // Move prev to the node immediately before the start of the reversal section

        for (int i = 0; i < left - 1; i++) {

            prev = prev.next;

        }

        

        // Start reversing the list

        ListNode current = prev.next;

        ListNode next;

        ListNode tail = current;  // The first node that will end up being the end of reversed section

        

        for (int i = left; i < right; i++) {

            next = current.next;

            current.next = prev.next;

            current = next;

        }

        

        // Connect the reversed section back to the rest of the list

        tail.next = current;

        

        return dummy.next;

    }

}

File: DataSets/DataSet-Old/LeetCodeTasksDebug/Fixed/92_reverse-linked-list-ii/solutions/c/solution.c
#include <stdio.h>

#include <stdlib.h>


struct ListNode* reverseBetween(struct ListNode* head, int left, int right) {

    if (left == right) return head;



    struct ListNode dummy;

    dummy.next = head;

    struct ListNode* prev = &dummy;



    // Advance to the node just before the 'left' position

    for (int i = 1; i < left; i++) {

        prev = prev->next;

    }



    // Reverse from 'left' to 'right'

    struct ListNode* curr = prev->next;

    struct ListNode* next = NULL;

    struct ListNode* lastUnreversed = curr;

    struct ListNode* reversedTail = prev->next;

    

    for (int i = left; i <= right; i++) {

        next = curr->next;

        curr->next = prev->next;

        prev->next = curr;

        curr = next;

    }



    // Connect the last part

    lastUnreversed->next = curr;



    return dummy.next;

}

